pkg/operator/resourcesynccontroller/resourcesync_controller.go

```diff
@@ -242,3 +244,53 @@ func (c *ResourceSyncController) eventHandler() cache.ResourceEventHandler {
 		DeleteFunc: func(obj interface{}) { c.queue.Add(controllerWorkQueueKey) },
 	}
 }
+
+func NewDebugHandler(controller *ResourceSyncController) http.Handler {
+	return &debugHTTPHandler{controller: controller}
+}
+
+type debugHTTPHandler struct {
+	controller *ResourceSyncController
+}
+
+type ResourceSyncRule struct {
+	Source      ResourceLocation `json:"source"`
+	Destination ResourceLocation `json:"destination"`
+}
+
+// ServeSyncRules provides a handler function to return the sync rules of the controller
+func (h *debugHTTPHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	syncRules := struct {
+		Secrets []ResourceSyncRule `json:"secrets"`
+		Configs []ResourceSyncRule `json:"configs"`
+	}{
+		[]ResourceSyncRule{},
+		[]ResourceSyncRule{},
+	}
+
+	h.controller.syncRuleLock.RLock()
+	defer h.controller.syncRuleLock.RUnlock()
+	syncRules.Secrets = append(syncRules.Secrets, mappedSyncRules(h.controller.secretSyncRules)...)
+	syncRules.Configs = append(syncRules.Configs, mappedSyncRules(h.controller.configMapSyncRules)...)
+
+	data, err := json.Marshal(syncRules)
+	if err != nil {
+		w.Write([]byte(err.Error()))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	w.Write(data)
+	w.WriteHeader(http.StatusOK)
+}
+
+func mappedSyncRules(syncRules map[ResourceLocation]ResourceLocation) []ResourceSyncRule {
```

Comment:
we should have a stable serialization, so sort the keys by namespace then by name.  Use a "normal" sort function.